components/FilterDropdown.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useState } from 'react';
+
+interface FilterDropdownProps {
+  name: string;
+  id: string;
+  value: string;
+  setStateAction: React.Dispatch<React.SetStateAction<string>>;
+  options: string[];
+  placeholder: string;
+}
+
+export default function FilterDropdown(props: FilterDropdownProps) {
```

Comment:
to keep things consistent, let's destructure the prop here! so it would be like: 
```
export default function FilterDropdown(
   {
      name,
      id,
      value,
      setStateAction,
      options,
      placeholder,
   }
    : FilterDropdownProps) { ... 
```
and later when u access props, you can directly use the prop, e.g. `setStateAction` instead of `props.setStateAction`